### DISCUSSION TITLE
Kitchenowl & Nextcloud OIDC Identity Provider

### DISCUSSION THREAD

**Discussion by gillesgagniard** [TomBursch/kitchenowl#406] [Q&A]
Hi everyone !

I'm currently trying to enable Kitchenowl authentication against Nextcloud as OIDC provider (through [OIDC Identity Provider app](https://github.com/H2CK/oidc)).

After configuring both sides (client ID and client secret, redirection URLs, ...) the authentication flow kind of start, but I end up with a generic 'Login failed' error in Nextcloud logs.

It really looks like https://github.com/H2CK/oidc/issues/339 ... Is Kitchenowl sending client credentials in the body of the POST token request, or is it using Basic Auth ? Is the latter, any way to pass them in the POST request body ?

Many thanks for your help !

**Comment by TomBursch** [comment]
I'm using the [pyoidc](https://pyoidc.readthedocs.io/en/latest/index.html) library for doing the token requests and yes, it looks like it is using Basic Auth by default. I'll change it to the POST method for the next update.
